rackcity/views/asset_views.py

```diff
@@ -164,49 +170,66 @@ def asset_add(request):  # need to make network and power connections here
     failure_message = ""
     if 'id' in data:
         failure_message += "Don't include id when adding an asset. "
-
+        return JsonResponse(
+            {"failure_message": failure_message},
+            status=HTTPStatus.BAD_REQUEST,
+        )
     serializer = AssetSerializer(data=data)
     if not serializer.is_valid(raise_exception=False):
         failure_message += str(serializer.errors)
-    if failure_message == "":
```

Comment:
A lot of this big chunk is just because I un-indented now that we're failing early.  Sorry it makes it hard to read